_episodes/02-keras.md

```diff
@@ -0,0 +1,514 @@
+> It is part of the tensorflow python package and can be imported using `from tensorflow import keras`.
+>
+> Keras includes functions, classes and definitions to define deep learning models, cost functions and
+> optimizers (optimizers are used to train a model).
+{:.callout}
+
+## 4. Choose a cost function and metrics
+Having selected the target enoding, we need to select an appropriate loss
+function that we will use during training.
+This loss function tells the training algorithm how wrong, or how 'far away' from the true
+value the predicted value is.
+
+For the one hot encoding we will use the Categorical Crossentropy loss.
+This is a measure for how close the distribution of the three neural network outputs corresponds
+to the distribution of the three values in the one hot encoding.
+It is lower if the distributions are more similar.
+In keras this is implemented in the `keras.losses.CategoricalCrossentropy` class.
+
+## 5. Choose a pretrained model or start building architecture from scratch
+Now we will build a neural network from scratch, and although this sounds like
+a daunting task, with Keras it is actually surprisingly straightforward.
+
+With keras you compose a neural network by creating layers and linking them
+together. For now we will only use one type of layer called a fully connected
+or Dense layer. In keras this is defined by the `keras.layers.Dense` class.
+
+A dense layer has a number of neurons, which is a parameter when you create the layer.
+When connecting the layer to its input and output layers every neuron in the dense
+layer gets an edge (i.e. connection) to ***all*** of the input neurons and ***all*** of the output neurons.
+The hidden layer in the image in the introduction of this episode is a Dense layer.
+
+The input in Keras also gets special treatment, Keras autmatically calculates the number of inputs
+and outputs a layer needs and therefore how many edges need to be created.
+This means we need to let Keras now how big our input is going to be.
+We do this by instantiating a `keras.Input` class and tell it how big our input is.
+
+~~~
+inputs = keras.Input(shape=training_data.shape[1])
+~~~
+{:.language-python}
+
+We store a reference to this input class in a variable so we can pass it to the creation of
+our hidden layer.
+Creating the hidden layer can then be done as follows:
+~~~
+hidden_layer = keras.layers.Dense(10, activation="relu")(inputs)
+~~~
+{:.language-python}
+
+The instantiation here has 2 parameters and a seemlingly strange combination of parenthenses, so
+let's take a closer look.
+The first parameter `10` is the number of neurons we want in this layer, this is one of the
+hyperparameters of our system and needs to be chosen carefully. We will get back to this in the section
+on hyperparameter tuning.
+The second parameter is the activation function to use, here we choose relu which is 0 for values
+0 and below and the same as the input for values above 0.
```

Comment:
```suggestion
The second parameter is the activation function to use, here we choose relu which is 0 for inputs that are
0 and below and the identity function (returning the same value) for inputs above 0.
```
And maybe add an image here?
Do we need to explain here why we need a non-linear activation function?